opensrp-app/src/main/java/org/smartregister/sync/helper/PractitionerSyncHelper.java

```diff
@@ -0,0 +1,96 @@
+    private HTTPAgent httpAgent;
+    private Gson gson;
+    protected static PractitionerSyncHelper instance;
+    private final AllSharedPreferences allSharedPreferences;
+    private static final String PRACTITIONER_URL = "/rest/practitioner";
+    private static final String PRACTITIONERS_BY_ID_AND_ROLE_URL = PRACTITIONER_URL + "/report-to";
+    private static final String PRACTITIONER_IDENTIFIER = "practitionerIdentifier";
+    private static final String PRACTITIONER_ROLE_CODE = "code";
+
+
+    public static PractitionerSyncHelper getInstance() {
+        if (instance == null) {
+            instance = new PractitionerSyncHelper(CoreLibrary.getInstance().context().getPractitionerRepository());
+        }
+        return instance;
+    }
+
+    private PractitionerSyncHelper(PractitionerRepository practitionerRepository) {
+        this.practitionerRepository = practitionerRepository;
+        this.allSharedPreferences = CoreLibrary.getInstance().context().allSharedPreferences();
+        this.httpAgent = CoreLibrary.getInstance().context().getHttpAgent();
+        this.gson = new Gson();
+    }
+
+
+    public void syncChwPractitionersByIdAndRoleFromServer() {
+        String practitionerIdentifier = allSharedPreferences.getUserPractitionerIdentifier();
+        String code = "Community Health Worker";
+        try {
+            String response = syncPractitioners(practitionerIdentifier, code);
+            List<Practitioner> practitioners = gson.fromJson(response, new TypeToken<List<Practitioner>>() {
+            }.getType());
+            for (Practitioner practitioner : practitioners) {
+                practitionerRepository.addOrUpdate(practitioner);
+            }
+        } catch (Exception ex) {
+            Timber.e(ex);
+        }
+    }
+
+    private String syncPractitioners(String practitionerIdentifier, String code) throws JSONException, NoHttpResponseException {
+        if (httpAgent == null) {
+            throw new IllegalArgumentException(PRACTITIONERS_BY_ID_AND_ROLE_URL + " HTTPAgent agent is null");
+        }
+
+        String baseUrl = getFormattedBaseUrl();
+
+        Response resp = httpAgent.fetch(
+                MessageFormat.format("{0}{1}{2}",
+                        baseUrl, PRACTITIONERS_BY_ID_AND_ROLE_URL,
+                        "?" + PRACTITIONER_IDENTIFIER + "=" + practitionerIdentifier +
+                                "&" + PRACTITIONER_ROLE_CODE + "=" + code));
+
+        if (resp.isFailure()) {
+            throw new NoHttpResponseException(PRACTITIONERS_BY_ID_AND_ROLE_URL + " not data returned");
+        }
+
+        return resp.payload().toString();
+    }
+
+    public String getFormattedBaseUrl() {
+        String baseUrl = CoreLibrary.getInstance().context().configuration().dristhiBaseURL();
+        String endString = "/";
+        if (baseUrl.endsWith(endString)) {
+            baseUrl = baseUrl.substring(0, baseUrl.lastIndexOf(endString));
+        }
+        return baseUrl;
+    }
```

Comment:
Does this method exist anywhere else? Should it be refactored placed in a Utility class for shared use?